front/lib/triggers/built-in-webhooks/linear/components/CreateWebhookLinearConnection.tsx

```diff
@@ -0,0 +1,174 @@
+
+  const { serviceData: linearData, isServiceDataLoading } =
+    useWebhookServiceData({
+      owner,
+      connectionId,
+      provider: "linear",
+    });
+
+  const { linearTeams, filteredTeams } = useMemo(() => {
+    const linearTeams = linearData?.teams ?? [];
+    const filteredTeams = linearTeams.filter((team) =>
+      `${team.name} (${team.key})`
+        .toLowerCase()
+        .includes(teamSearchQuery.toLowerCase())
+    );
+    return { linearTeams, filteredTeams };
+  }, [linearData, teamSearchQuery]);
+
+  const teamsInDropdown = useMemo(
+    () =>
+      filteredTeams.filter(
+        (team) => !selectedTeams.some((t) => t.id === team.id)
+      ),
+    [filteredTeams, selectedTeams]
+  );
+
+  useEffect(() => {
+    const isReady = !!(connectionId && selectedTeams.length > 0);
+
+    if (isReady && onDataToCreateWebhookChange) {
+      onDataToCreateWebhookChange({
+        connectionId,
+        remoteMetadata: {
+          teams: selectedTeams,
+        },
+      });
+    } else if (onDataToCreateWebhookChange) {
+      onDataToCreateWebhookChange(null);
+    }
+
+    if (onReadyToSubmitChange) {
+      onReadyToSubmitChange(isReady);
+    }
+  }, [
+    connectionId,
+    selectedTeams,
+    onDataToCreateWebhookChange,
+    onReadyToSubmitChange,
+  ]);
+
+  const handleAddTeam = (team: LinearTeam) => {
+    if (!selectedTeams.some((t) => t.id === team.id)) {
+      setSelectedTeams([...selectedTeams, team]);
+    }
+    setTeamSearchQuery("");
+    setShowTeamDropdown(false);
+  };
+
+  const handleRemoveTeam = (team: LinearTeam) => {
+    setSelectedTeams(selectedTeams.filter((t) => t.id !== team.id));
+  };
+
+  return (
+    <div className="space-y-6">
+      {isServiceDataLoading ? (
+        <div className="mt-2 flex items-center gap-2 py-2">
+          <Spinner size="sm" />
+          <span className="text-sm text-muted-foreground">
+            Loading teams...
+          </span>
+        </div>
+      ) : (
+        <div className="space-y-4">
+          <div>
+            <Label>
+              Teams{" "}
+              {selectedTeams.length === 0 && (
+                <span className="text-warning">*</span>
+              )}
+            </Label>
+            <div className="mt-2 flex flex-col gap-2">
+              <div className="flex flex-wrap items-center gap-1">
+                {selectedTeams.map((team) => (
+                  <Chip
+                    key={team.id}
+                    size="xs"
+                    label={`${team.name} (${team.key})`}
+                    color="primary"
+                    className="m-0.5"
+                    onRemove={() => handleRemoveTeam(team)}
+                  />
+                ))}
+              </div>
+              {linearTeams.length > 0 && (
+                <div className="flex">
+                  <DropdownMenu
+                    open={showTeamDropdown}
+                    onOpenChange={setShowTeamDropdown}
+                  >
```

Comment:
Why do we need to maintain the state? Let's just use the trigger and let radix handle the magic no?